src/pkg/egress/syslog/https.go

```diff
@@ -32,15 +42,87 @@ func NewHTTPSWriter(
 ) egress.WriteCloser {
 
 	client := httpClient(netConf, tlsConf)
+	if binding.URL.Query().Get("batching") == "true" {
+		return &BatchHTTPSWriter{
+			HTTPSWriter: HTTPSWriter{
+				url:             binding.URL,
+				appID:           binding.AppID,
+				hostname:        binding.Hostname,
+				client:          client,
+				egressMetric:    egressMetric,
+				syslogConverter: c,
+			},
+			msgBatch:     "",
+			batchSize:    BATCHSIZE,
+			sendInterval: time.Second,
+		}
+	} else {
+		return &HTTPSWriter{
+			url:             binding.URL,
+			appID:           binding.AppID,
+			hostname:        binding.Hostname,
+			client:          client,
+			egressMetric:    egressMetric,
+			syslogConverter: c,
+		}
+	}
+}
+
+func (w *BatchHTTPSWriter) sendMsgBatch() error {
+	req := fasthttp.AcquireRequest()
+	req.SetRequestURI(w.url.String())
+	req.Header.SetMethod("POST")
+	req.Header.SetContentType("text/plain")
+	req.SetBodyString(w.msgBatch)
+
+	w.msgBatch = ""
+	w.sendTimer = nil
+
+	resp := fasthttp.AcquireResponse()
+
+	err := w.client.Do(req, resp)
+	if err != nil {
+		return w.sanitizeError(w.url, err)
+	}
+
+	if resp.StatusCode() < 200 || resp.StatusCode() > 299 {
+		return fmt.Errorf("syslog Writer: Post responded with %d status code", resp.StatusCode())
+	}
+
+	return nil
+}
 
-	return &HTTPSWriter{
-		url:             binding.URL,
-		appID:           binding.AppID,
-		hostname:        binding.Hostname,
-		client:          client,
-		egressMetric:    egressMetric,
-		syslogConverter: c,
+// Modified Write function
+func (w *BatchHTTPSWriter) Write(env *loggregator_v2.Envelope) error {
+	msgs, err := w.syslogConverter.ToRFC5424(env, w.hostname)
+	if err != nil {
+		return err
+	}
+
+	for _, msg := range msgs {
+		if len(w.msgBatch) == 0 {
+			w.msgBatch = string(msg)
+		} else {
+			w.msgBatch += string(msg)
+		}
+		w.egressMetric.Add(1)
 	}
+
+	if w.sendTimer == nil {
+		w.sendTimer = time.AfterFunc(w.sendInterval, func() {
```

Comment:
Why do this here and have `Stop` below rather than just always trying to call it?